### ISSUE TITLE
Uncaught TypeError: Cannot read property 'type' of undefined

### ISSUE THREAD

**Issue by tmarrs** [slang800/atom-tidy-markdown#5]
[Enter steps to reproduce below:]
1. ...
2. ...

**Atom Version**: 0.187.0
**System**: Mac OS X 10.10.2
**Thrown From**: [tidy-markdown](https://github.com/slang800/atom-tidy-markdown) package, v0.1.1
### Stack Trace

Uncaught TypeError: Cannot read property 'type' of undefined

```
At /Applications/Atom.app/Contents/Resources/app/src/pane.js:784

TypeError: Cannot read property 'type' of undefined
  at preprocessAST (/Users/tmarrs/.atom/packages/tidy-markdown/node_modules/tidy-markdown/lib/index.coffee:102:18)
  at preprocessAST (/Users/tmarrs/.atom/packages/tidy-markdown/node_modules/tidy-markdown/lib/index.coffee:114:7)
  at module.exports (/Users/tmarrs/.atom/packages/tidy-markdown/node_modules/tidy-markdown/lib/index.coffee:221:9)
  at TidyMarkdown.run (/Users/tmarrs/.atom/packages/tidy-markdown/lib/tidy-markdown.coffee:36:17)
  at /Users/tmarrs/.atom/packages/tidy-markdown/lib/tidy-markdown.coffee:23:12
  at History.module.exports.History.transact (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/history.js:115:18)
  at TextBuffer.module.exports.TextBuffer.transact (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:687:27)
  at /Users/tmarrs/.atom/packages/tidy-markdown/lib/tidy-markdown.coffee:21:14
  at TextBuffer.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/emissary/lib/emitter.js:118:15)
  at TextBuffer.module.exports.TextBuffer.saveAs (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:968:12)
  at TextBuffer.module.exports.TextBuffer.save (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:958:19)
  at TextEditor.module.exports.TextEditor.save (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:621:26)
  at Pane.module.exports.Pane.saveItem (/Applications/Atom.app/Contents/Resources/app/src/pane.js:542:18)
  at Pane.module.exports.Pane.saveActiveItem (/Applications/Atom.app/Contents/Resources/app/src/pane.js:525:19)
  at Workspace.module.exports.Workspace.saveActivePaneItem (/Applications/Atom.app/Contents/Resources/app/src/workspace.js:689:35)
  at atom-workspace.atom.commands.add.core:save (/Applications/Atom.app/Contents/Resources/app/src/workspace-element.js:295:30)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app/src/command-registry.js:246:29)
  at /Applications/Atom.app/Contents/Resources/app/src/command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:558:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:396:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/Applications/Atom.app/Contents/Resources/app/src/window-event-handler.js:182:20)

```
### Commands

```
 12x -9:41.6 core:undo (ol.tree-view.full-menu.list-tree.has-collapsable-children.focusable-panel)
     -9:17.9 core:save (atom-text-editor.editor)
     -8:52.7 markdown-preview:toggle (atom-text-editor.editor)
  7x -8:15.6 core:save (ol.tree-view.full-menu.list-tree.has-collapsable-children.focusable-panel)
     -5:51.2 core:confirm (atom-text-editor.editor.mini)
     -1:38.3 core:copy (atom-text-editor.editor.is-focused)
     -1:35.3 core:paste (atom-text-editor.editor.is-focused)
     -1:26.9 core:backspace (atom-text-editor.editor.is-focused)
     -1:09.4 core:move-left (atom-text-editor.editor.is-focused)
     -1:08.3 core:paste (atom-text-editor.editor.is-focused)
  6x -1:07.1 core:backspace (atom-text-editor.editor.is-focused)
  2x -0:51.8 core:paste (atom-text-editor.editor.is-focused)
     -0:02.6 core:save (atom-text-editor.editor.is-focused)
```
### Config

``` json
{
  "core": {
    "disabledPackages": [
      "language-ruby-on-rails",
      "minimap",
      "wrap-guide"
    ],
    "themes": [
      "atom-dark-ui",
      "monokai"
    ]
  }
}
```
### Installed Packages

``` coffee
# User
angularjs, v0.2.0
angularjs-styleguide-snippets, v0.5.0
asciidoc-preview, v0.4.3
atom-beautify, v0.23.0
atom-lint, v0.20.1
atom-runner, v2.3.0
autoclose-html, v0.15.0
autocomplete-html, v0.5.0
coffee-lint, v0.7.3
color-picker, v1.5.0
esformatter, v1.4.0
eslint, v0.15.0
git-projects, v1.11.0
js-refactor, v0.3.0
jsformat, v0.7.18
jsonlint, v1.0.2
language-asciidoc, v0.11.1
language-javascript-semantic, v0.1.0
linter, v0.12.0
linter-csslint, v0.0.11
linter-eslint, v0.5.0
linter-htmlhint, v0.0.8
linter-javac, v0.1.3
linter-rubocop, v0.2.2
linter-tidy, v1.0.0
markdown-format, v2.5.0
markdown-table-formatter, v2.5.2
mocha, v0.0.5
monokai, v0.12.0
open-last-project, v0.3.5
pretty-json, v0.3.2
project-manager, v1.15.5
refactor, v0.4.1
regex-railroad-diagram, v0.7.1
ruby-block, v0.3.3
ruby-block-converter, v3.3.0
script, v2.18.0
slash-closer, v0.7.1
tidy-markdown, v0.1.1

# Dev
No dev packages
```

**Comment by notslang** [comment]
Could you provide the Markdown content that caused this error? And are you able to reproduce this consistently with a given file?

That info would be very helpful in debugging this.

**Comment by yantongl** [comment]
In addition, when this happens, the file failed to save. I have to disable tidy-markdown so I can save my change. Could we make sure files will always be saved, even tidy-markdown failed to fix the file?

**Comment by notslang** [comment]
@yantongl what version are you using, and what is the markdown you're working with?

**Comment by yantongl** [comment]
atom version 0.188.0
tidy-markdown version 0.2.0
OS Windows 7, SP1

This is the simple fake file I used to repro this problem:

```

# tests
+ adas
  - asdas
  - sdas

+ sdfsdas
    - sddfasdfdfs
    - sdfafasdfsa  
+ asdfads

```

**Comment by notslang** [comment]
Thanks @yantongl! I was able to reproduce the issue with your markdown, and fixed it in the v0.2.2 release. :cake:
